rename_wpt/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/google/go-github/v35/github"
+	"github.com/web-platform-tests/wpt-metadata/go_util"
+)
+
+// This main() gets all the WPT test renaming and deletions between two WPT SHAs, then it deletes/renames
+// appropriately in the wpt-metadata repository. This program will roll in the latest WPT SHA once a day
+// through a cron job.
+func main() {
+	// TODO(Kyleju): plumb the before, after SHA through shell. Update SHA through shell as well.
+	shaAfter := "e021a4c8f92596f3261e2d7540d017f5138deb68"
+	shaBefore := "012651bc39bb45a6dc9c93a968b47dd9310a1e8a"
+
+	// Plumb the SHA information from a text file, WPT SHA rolls on a daily basis
+	renames, deletes := getChangesBetweenSHAs(context.Background(), shaBefore, shaAfter)
+	if renames == nil {
+		return
+	}
+
+	fmt.Println("Renaming WPT tests...")
+	for before, after := range renames {
+		renameMetadata(before, after)
+	}
+
+	fmt.Println("Deleting WPT tests...")
+	for _, delete := range deletes {
+		go_util.DeleteTestFromMetadata(delete)
+	}
+}
+
+// renameMetadata deletes the before WPT test and adds the after WPT test, if exists.
+func renameMetadata(before, after string) {
+	metadataLinks := go_util.DeleteTestFromMetadata(before)
+	if len(metadataLinks) == 0 {
+		return
+	}
+
+	afterTestname := path.Base(after)
+	// Rename the before test name to the after test name.
+	for linkIndex, link := range metadataLinks {
+		for resultIndex := range link.Results {
+			fmt.Println("Renaming", before, after)
+			metadataLinks[linkIndex].Results[resultIndex].TestPath = afterTestname
+		}
+	}
+
+	for _, link := range metadataLinks {
+		go_util.WriteMetadataLink(after, link)
+	}
+}
+
+// getChangesBetweenSHAs returns all the renamed and deleted files between two WPT SHAs.
+// TODO(kyleju): In WPT tests, if we have a foo.any.js file, that would have tests
+// foo.any.html, foo.any.worker.html and foo.any.sharedworker.html. To perfectly rename/map all tests,
+// we need MANIFEST informaiton.
+func getChangesBetweenSHAs(ctx context.Context, shaBefore, shaAfter string) (map[string]string, []string) {
+	fmt.Println("Getting WPT changes bewtween", shaBefore, "and", shaAfter)
+	if shaBefore == shaAfter {
+		fmt.Println("No new SHA after", shaBefore)
```

Comment:
This doesn't really match the condition. Assuming this is possible and shouldn't be a fatal error, maybe "Nothing to do since the before/after SHAs are the same"?

Reply:
It is possible once we start rolling the SHA on a daily basis like the manifest but I changed the message.